endpoints/openrtb2/auction.go

```diff
@@ -1980,6 +2007,65 @@ func getAccountID(pub *openrtb2.Publisher) string {
 	return metrics.PublisherUnknown
 }
 
+// Returns the account from the request
+func getAccountIdFromRawRequest(hasStoredRequest bool, storedRequest json.RawMessage, originalRequest []byte) (string, bool, []error) {
+	var accountId string
+	var isAppReq bool
+	var err error
+
+	request := originalRequest
+	if hasStoredRequest {
+		request = storedRequest
```

Comment:
It seems very possible that the original request may contain the account identifying data, and the stored request may not, and just be used to hold the bidder parameters and perhaps some other non-account things. There is nothing in the spec to say if you are using a stored request, you must store as much as possible in the request.

You should check the original request if the stored request does not have the account identifier.